services/graph.js

```diff
@@ -1,136 +1,230 @@
-import 'isomorphic-fetch';
-
-export default class GraphService {
-  constructor(options) {
-    this.opts = options;
-  }
-
-  /**
-   * Get all tags
-   */
-  getAllTags() {
-    return fetch(
-      `${process.env.WRI_API_URL}/graph/query/list-concepts?application=${process.env.APPLICATIONS}&env=${process.env.API_ENV}`,
-      { headers: { 'Upgrade-Insecure-Requests': 1 } }
-
-    )
-      .then(response => response.json())
-      .then(response => response.data);
-  }
-  /**
-   * Get inferred tags
-   */
-  getInferredTags(tags) {
-    return fetch(
-      `${process.env.WRI_API_URL}/graph/query/concepts-inferred?concepts=${tags}&application=${process.env.APPLICATIONS}&env=${process.env.API_ENV}`,
-      { headers: { 'Upgrade-Insecure-Requests': 1 } }
-
-    )
-      .then(response => response.json())
-      .then(response => response.data);
+import WRISerializer from 'wri-json-api-serializer';
+
+// utils
+import { WRIAPI } from 'utils/axios';
+import { logger } from 'utils/logs';
+
+/**
+ * Get all tags
+ */
+export const fetchAllTags = (params = {}) => {
+  logger.info('Fetch all tags');
+  return WRIAPI.get('graph/query/list-concepts',
+    {
+      headers: { 'Upgrade-Insecure-Requests': 1 },
+      params: {
+        env: process.env.API_ENV,
+        application: process.env.APPLICATIONS,
+        ...params
+      }
+    })
+    .then(response => response.data.data)
+    .catch((response) => {
+      const { status, statusText } = response;
+      logger.error(`Error fetching all tags: ${status}: ${statusText}`);
+      throw new Error(`Error fetching all tags: ${status}: ${statusText}`);
+    });
+};
+
+/**
+ * Get inferred tags
+ */
+export const fetchInferredTags = (tags, params = {}) => {
+  logger.info('Fetch inferred tags');
+  return WRIAPI.get(`graph/query/concepts-inferred?concepts=${tags}`,
+    {
+      headers: { 'Upgrade-Insecure-Requests': 1 },
+      params: {
+        env: process.env.API_ENV,
+        application: process.env.APPLICATIONS,
+        ...params
+      }
+    })
+    .then(response => response.data.data)
+    .catch((response) => {
+      const { status, statusText } = response;
+      logger.error(`Error fetching inferred tags: ${tags} ${status}: ${statusText}`);
+      throw new Error(`Error inferred tags: ${tags} ${status}: ${statusText}`);
+    });
+};
+
+/**
+* Get dataset tags
+*/
+export const fetchDatasetTags = (datasetId, params = {}) => {
```

Comment:
I think all tag-related endpoints should go to the `dataset` service. All endpoints start with `/dataset` which is basically the way we organize the services: under the same root endpoint. And this case is not different, we are updating the tags of a dataset.